mitodl/open-discussions
# Split each OCW run into its own course

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #3243 

#### What's this PR do?
Adds a management command to split each OCW run into its own course.  Modifies OCW import functions to do the same.

#### How should this be manually tested?
- On the master branch, import the following if you don't already have it:
```
from course_catalog.api import *
sync_ocw_courses(course_prefixes=[
"PROD/18/18.01/Fall_2003/18-01-single-variable-calculus-fall-2003/",
"PROD/18/18.01/Fall_2005/18-01-single-variable-calculus-fall-2005/",
"PROD/18/18.01/Fall_2006/18-01-single-variable-calculus-fall-2006/",
], blocklist=[], force_overwrite=True, upload_to_s3=True)
```
- Search for "18.01" in open discussions, there should be 1 result for "Single Variable Calculus" and it should have 2 published runs.
- Run `python manage.py split_ocw_courses`
- Run `python manage.py recreate_index`
- Search for "18.01" in open discussions, there should be 2 results for "Single Variable Calculus" and they should each have 1 published run.  Thumbnails should be different.

- Run `backpopulate_ocw_courses` for a little while, or run the `sync_ocw_courses` function above again on a subset of prefixes, and make sure it works correctly
